pkg/frontend/querymiddleware/results_cache.go

```diff
@@ -519,6 +529,15 @@ func partitionCacheExtents(req MetricsQueryRequest, extents []Extent, minCacheEx
 		requests = append(requests, req)
 	}
 
+	// Track samples processed from cache
+	var cachedSamplesProcessed uint64 = 0
+	for _, stepStat := range cachedPerStepStat {
+		cachedSamplesProcessed += uint64(stepStat.Value)
+	}
+	if details := QueryDetailsFromContext(ctx); details != nil {
+		details.SamplesProcessedFromCache = cachedSamplesProcessed
+	}
```

Comment:
Design-wise, I don't think it's `partitionCacheExtents()` responsibility taking the context in input and updating it.